linux/LinuxProcessList.c

```diff
@@ -486,6 +496,62 @@ static bool LinuxProcessList_readStatmFile(LinuxProcess* process, const char* di
    return (errno == 0);
 }
 
+static bool LinuxProcessList_readSmapsFile(LinuxProcess* process, const char* dirname, const char* name, bool haveSmapsRollup) {
+   //http://elixir.free-electrons.com/linux/v4.10/source/fs/proc/task_mmu.c#L719
+   //kernel will return data in chunks of size PAGE_SIZE or less.
```

Comment:
@hishamhm: True and well analyzed! However, such reads will never happen for regular `smaps` because data is returned in chunks of PAGE_SIZE. For `smaps_rollup`, as you have correctly pointed out, this discussion is irrelevant anyways.
Regarding the verification of this claim: The linked code does appear to write out each result entry separately and when I run `strace cat /proc/$$/smaps` each read returns as many entries as fit into a 4k-page, without any “half-entries”. (For the most part we have to trust the previous author's judgment here. Fortunately however we do not need to worry about kernel changes, since we're only really talking about Linux ~2.6.26 up to 2.13 here).